### PR TITLE
Run linter for each contribution

### PR DESCRIPTION
**Pull request by VysotskiVadim**
### Description
Implementation of #4085.

The PR also contains fixed for existing issues.
Follow up issues which could be fixed later: #5314

### CODE REVIEW

File: libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
-      android:top="4dp" />
-
-  <corners android:radius="5dp" />
-</shape>
\ No newline at end of file
+    <stroke android:width="1dp" android:color="@android:color/black"/>
+    <solid android:color="@android:color/white"/>
+    <padding android:left="4dp" android:right="4dp" android:bottom="4dp" android:top="4dp"/>
+    <corners android:radius="5dp"/>
+</shape>
```
Comment (Zayankovsky) [inline libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml]: This seems unrelated. Why do we need these changes?
They contradict Android Studio formatter (Code -> Reformat Code).

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_right.xml [hunk 2]
```diff
@@ -1,9 +1,3 @@
-<vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:height="20dp"
-    android:viewportHeight="20"
-    android:viewportWidth="20"
-    android:width="20dp">
-  <path
-      android:fillColor="?attr/maneuverTurnIconColor"
-      android:pathData="M15,10a2,2,0,1,1,2,2a2,2,0,0,1-2-2Zm-8,1.0549300000000006h1v-.009090000000000487a.35757.35757,0,0,1,.37738.54242l-1.3437500000000009,2.9518400000000007a.3866.3866,0,0,0,.62354.40942l6.34283-4.903680000000001l-6.34283-4.90369a.38659.38659,0,0,0-.62354.40942l1.3437500000000009,2.9518500000000003a.35757.35757,0,0,1-.3773800000000005.5424199999999999v.009090000000000487h-.9918199999999997l.00011999999999989797-.020080000000000098l-.011.00183a6.05607,6.05607,0,0,0-5.98345,5.9633199999999995h-.013849999999999696v1.48767a.51232.51232,0,0,0,.51233.5123299999999986h.9753400000000001a.51232.51232,0,0,0,.51233-.5123299999999986v-1.3777500000000007h.008300000000000196a4.05782,4.05782,0,0,1,3.98877-4.07324" />
```
Comment (VysotskiVadim) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_right.xml:8]: Android Lint says that `.009` in the pathData can cause a crash on some devices. Lint suggests replace `.009` by `0.009`. I tried to fix the errors manually but the found out that we have too many vector drawables.

I used the avocado tool which I found in the [article](https://android-blogs.medium.com/optimize-svgs-remove-lint-warning-of-the-long-vector-path-fe0b34f80782) to automatically fix all paths. Unfortunately avocado also reformats the files.

Comment (VysotskiVadim) [reply libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_right.xml:8]: I will try to find a way to configure avocado or use different tool to fix paths.

File: libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml (hunk 1 above)
Comment (VysotskiVadim) [reply libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml]: See [the comment](https://github.com/mapbox/mapbox-navigation-android/pull/5312/files#r774981365)

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt [hunk 3]
```diff
@@ -128,6 +131,7 @@ class MapboxReplayer {
      * When initializing or testing an app, it is needed and useful to push a device location.
      * This helper function can be used to push the actual location into the replayer.
      */
+    @RequiresPermission(anyOf = [ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION])
```
Comment (VysotskiVadim) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt:134]: The Kotlin compiler doesn't fail a build because of the `RequiresPermission` annotation. But Android Lint does. I'm wondering, is this a breaking change? Probably no. 🤔 
@mapbox/navigation-android , what do you think?

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml [hunk 4]
```diff
@@ -1,9 +1,3 @@
...
-    android:height="20dp"
-    android:viewportHeight="20"
-    android:viewportWidth="20"
-    android:width="20dp">
-  <path
-      android:fillColor="?attr/maneuverTurnIconColor"
-      android:pathData="M3,12a2,2,0,1,1,2-2a2,2,0,0,1-2,2Zm10.00293-.96332a4.05782,4.05782,0,0,1,3.98877,4.07324h.008300000000001972v1.37775a.51232.51232,0,0,0,.5123299999999986.5123299999999986h.97534a.51232.51232,0,0,0,.5123300000000022-.5123299999999986v-1.4876700000000014h-.013850000000001472a6.05607,6.05607,0,0,0-5.9834-5.96332l-.011-.00183l.00012.02008h-.9918700000000005v-.009090000000000487a.35757.35757,0,0,1-.37738-.54242l1.34375-2.95184a.38659.38659,0,0,0-.62354-.40942l-6.342829999999999,4.90368l6.34283,4.90369a.3866.3866,0,0,0,.62354-.40942l-1.34375-2.95184a.35757.35757,0,0,1,.3773800000000005-.5424299999999977v.00909h1" />
-</vector>
\ No newline at end of file
+<vector xmlns:android="http://schemas.android.com/apk/res/android" android:height="20dp" android:viewportHeight="20" android:viewportWidth="20" android:width="20dp">
```
Comment (RingerJK) [inline libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml:1]: it became less human-readable. Could you revert this

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/MapboxReplayer.kt:134]: I don't think it's breaking changes if you can still build an app with `./gradlew build`

File: libnavigation-core/build.gradle [hunk 5]
```diff
@@ -30,6 +30,13 @@ android {
         buildConfigField 'String', "NAV_NATIVE_SDK_VERSION", String.format("\"%s\"", project.ext.version.mapboxNavigator)
     }
 
+    lintOptions {
+        // the NullSafeMutableLiveData rule fails with index out of bounds exception analysing the MapboxNavigationSDK.kt.
+        // Try removing the line after updating androidx.lifecycle libraries
+        disable "NullSafeMutableLiveData"
```
Comment (RingerJK) [inline libnavigation-core/build.gradle:36]: what do you think if we move rules to `lint.xml` file and keep `gradle` file cleaner

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_right.xml (hunk 2 above)
Comment (VysotskiVadim) [reply libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_right.xml:8]: fixed paths using regex

File: libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml (hunk 1 above)
Comment (VysotskiVadim) [reply libnavui-maneuver/src/main/res/drawable/mapbox_exit_board_background.xml]: fixed

File: libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml (hunk 4 above)
Comment (VysotskiVadim) [reply libnavui-maneuver/src/main/res/drawable/mapbox_ic_arrive_left.xml]: fixed

File: libnavigation-core/build.gradle (hunk 5 above)
Comment (VysotskiVadim) [reply libnavigation-core/build.gradle]: for some reason, I don't know why, `NullSafeMutableLiveData` stopped crashing Lint 😮

Comment (VysotskiVadim) [reply libnavigation-core/build.gradle:36]: Can we disable a rule in `lint.xml` file so that it doesn't run? I found only examples of ignoring issues https://developer.android.com/studio/write/lint#sample-lint.xml-file

Comment (Zayankovsky) [reply libnavigation-core/build.gradle:36]: What is the difference between disabling the rule and ignoring it?
I tried locally with ignoring `NullSafeMutableLiveData` in lint.xml file and it worked.

File: libnavigation-base/src/main/res/values-uz/strings.xml [hunk 6]
```diff
@@ -2,7 +2,8 @@
 
     <!-- Time formatting -->
     <plurals name="mapbox_number_of_days">
-        <item quantity="other">kun</item>
+        <item quantity="one">kun</item>
+        <item quantity="other">kunlar</item>
```
Comment (Zayankovsky) [inline libnavigation-base/src/main/res/values-uz/strings.xml:6]: Are you sure about this? Both Yandex and Google translate "twenty days" as "yigirma **kun**".

Comment (VysotskiVadim) [reply libnavigation-base/src/main/res/values-uz/strings.xml:6]: thanks, let me double check it 👍

File: libnavigation-core/build.gradle (hunk 5 above)
Comment (VysotskiVadim) [reply libnavigation-core/build.gradle]: @Zayankovsky , you're right, it disabling and ignoring means the same in this context. Fixed, thanks 👍

File: libnavigation-base/src/main/res/values-uz/strings.xml (hunk 6 above)
Comment (VysotskiVadim) [reply libnavigation-base/src/main/res/values-uz/strings.xml:6]: great catch! "days" is translated like "kunlar" but "20 days" is "20 kun". It's better not to assume that other languages works like the English and translate in the context 😅

File: libnavigation-base/src/main/res/values-uz/strings.xml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
     <!-- Time formatting -->
     <plurals name="mapbox_number_of_days">
-        <item quantity="other">kun</item>
```
Comment (Zayankovsky) [inline libnavigation-base/src/main/res/values-uz/strings.xml:5]: Looks like we need `other` as well, because I get the following crash, when the trip takes longer than one day:
```
android.content.res.Resources$NotFoundException: Plural resource ID #0x7f0d0002 quantity=8 item=other
    at android.content.res.ResourcesImpl.getQuantityText(ResourcesImpl.java:326)
    at android.content.res.Resources.getQuantityText(Resources.java:455)
    at android.content.res.Resources.getQuantityString(Resources.java:558)
    at com.mapbox.navigation.base.internal.time.TimeFormatter.formatDays(TimeFormatter.kt:108)
    at com.mapbox.navigation.base.internal.time.TimeFormatter.formatTimeRemaining(TimeFormatter.kt:94)
    at com.mapbox.navigation.ui.tripprogress.model.TimeRemainingFormatter.format(TimeRemainingFormatter.kt:31)
    at com.mapbox.navigation.ui.tripprogress.model.TimeRemainingFormatter.format(TimeRemainingFormatter.kt:17)
    at com.mapbox.navigation.ui.tripprogress.model.TripProgressUpdateFormatter.getTimeRemaining(TripProgressUpdateFormatter.kt:115)
    at com.mapbox.navigation.ui.tripprogress.view.MapboxTripProgressView.render(MapboxTripProgressView.kt:89)
    at com.mapbox.navigation.examples.core.MapboxNavigationActivity.routeProgressObserver$lambda-7(MapboxNavigationActivity.kt:229)
    at com.mapbox.navigation.examples.core.MapboxNavigationActivity.$r8$lambda$4EdaMIwhYKIMRWmmoWh2laPioJo(Unknown Source:0)
    at com.mapbox.navigation.examples.core.MapboxNavigationActivity$$ExternalSyntheticLambda3.onRouteProgressChanged(Unknown Source:2)
    at com.mapbox.navigation.core.trip.session.MapboxTripSession.updateRouteProgress(MapboxTripSession.kt:629)
    at com.mapbox.navigation.core.trip.session.MapboxTripSession.access$updateRouteProgress(MapboxTripSession.kt:58)
    at com.mapbox.navigation.core.trip.session.MapboxTripSession$navigatorObserver$1$onStatus$1.invokeSuspend(MapboxTripSession.kt:314)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
    at android.os.Handler.handleCallback(Handler.java:883)
    at android.os.Handler.dispatchMessage(Handler.java:100)
    at android.os.Looper.loop(Looper.java:224)
    at android.app.ActivityThread.main(ActivityThread.java:7590)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```

**Comment by VysotskiVadim** [comment]
Tested changes of all vector drawables in the Android Studio previewer. Let me know if you think this is not enough 🙂

**Review by Zayankovsky — APPROVED** [review]

**Review by LukasPaczos — APPROVED** [review]
